topics/go/algorithms/slices/max/max.go

```diff
@@ -1,24 +1,25 @@
 package slices
 
-// Max returns the maximum integer in the slice.
-func Max(n []int) int {
+import (
+	"fmt"
+)
 
-	// If the length of the slice is 1 then return the integer at index 0.
-	if len(n) == 1 {
-		return n[0]
+// Max returns the maximum integer in the slice.
+func Max(n []int) (int, error) {
+	if len(n) == 0 {
+		return 0, fmt.Errorf("slice %#v has no elements", n)
 	}
 
-	// Create an integer and assign it to the first index of the slice.
 	max := n[0]
```

Comment:
What do you think about moving this just before the loop on line 16 but under the command? This line goes with the range to find the min. This will reduce a few extra lines of whitespace.